numpy/linalg/umath_linalg.c.src

```diff
@@ -2446,20 +2458,23 @@ init_@lapack_func@(GESDD_PARAMS_t *params,
     npy_uint8 *mem_buff = NULL;
     npy_uint8 *mem_buff2 = NULL;
     npy_uint8 *a, *s, *u, *vt, *work, *iwork;
-    size_t a_size = (size_t)m*(size_t)n*sizeof(@ftyp@);
+    size_t safe_m = m;
+    size_t safe_n = n;
+    size_t a_size = safe_m * safe_n * sizeof(@ftyp@);
     fortran_int min_m_n = m<n?m:n;
-    size_t s_size = ((size_t)min_m_n)*sizeof(@ftyp@);
+    size_t safe_min_m_n = min_m_n;
+    size_t s_size = safe_min_m_n * sizeof(@ftyp@);
     fortran_int  u_row_count, vt_column_count;
     size_t u_size, vt_size;
     fortran_int work_count;
     size_t work_size;
-    size_t iwork_size = 8*((size_t)min_m_n)*sizeof(fortran_int);
+    size_t iwork_size = 8 * safe_min_m_n * sizeof(fortran_int);
 
     if (!compute_urows_vtcolumns(jobz, m, n, &u_row_count, &vt_column_count))
         goto error;
 
-    u_size = ((size_t)u_row_count)*m*sizeof(@ftyp@);
-    vt_size = n*((size_t)vt_column_count)*sizeof(@ftyp@);
+    u_size = ((size_t)u_row_count) * safe_m * sizeof(@ftyp@);
```

Comment:
Why casting here? Wouldn't `safe_u_row_count` and `safe_vt_col_count` be more consistent choices?